install.py

```diff
@@ -5,243 +5,287 @@
+        self.ncurses()
+        self.install_mojo()
+        self.rc_path()
+        self.handle_rc()
+
+    def hasstr(self, target: str, original: str) -> bool:
+        if not isinstance(target, str) or not isinstance(original, str):
+            return False
+
+        elif len(target) > len(original):
+            return False
+        
+        else:
+            start = 0
+            stop = len(target)
+
+            while start < len(original):
+                if original[start:stop] == target:
+                    return True
+                start += 1
+                stop += 1
+
+            else:
+                return False
+
+    def _help(self) -> int:
+        sys.stdout.write("Usage: python3 install.py [options]\n")
+        sys.stdout.write("Options:\n")
+        sys.stdout.write("  --dir=<path>  | -d=<path>  : Set the working directory\n")
+        sys.stdout.write("  --global      | -g         : Install the libs globally\n")
+        sys.stdout.write("  --help        | -h         : Show this help message\n")
+        sys.stdout.write("  --mojo        | -m         : Only install mojo (modular must be installed)\n")
+        sys.stdout.write("  --fedora      | -f         : Install for fedora\n")
+        sys.stdout.write("  --modular-token <token>    : Set the modular token\n")
+        return 0
+
+    def is_authenticated(self) -> None:
+        if self.modular:
+            result = subprocess.run(["modular", "config-list"], capture_output=True).stdout.decode("utf-8")
+            self.authenticated = self.hasstr("user.id", result)
+
+    def handle_args(self) -> None:
+        for arg in self.args:
+            if self.skip_next_arg:
+                self.skip_next_arg = False
+                continue
+
+            if arg.startswith("--dir="):
+                self.working_dir = arg.split("=")[1]
+            elif arg.startswith("-d="):
+                self.working_dir = arg.split("=")[1]
+            elif arg == "--global":
+                self.install_global = True
+            elif arg == "-g":
+                self.install_global = True
+            elif arg == "--mojo":
+                self.onlyMojo = True
+            elif arg == "-m":
+                self.onlyMojo = True
+            elif arg == "--fedora":
+                self.fedora = True
+            elif arg == "-f":
+                self.fedora = True
+            elif arg == "--modular-token":
+                index = self.args.index(arg) + 1
+                
+                if index >= len(self.args):
+                    sys.stdout.write("\nNo token provided")
+                    exit(1)
+
+                if self.token is not None:
+                    if self.token == "" or not self.token.startswith("mut_") or not len(self.token) == 36:
+                        sys.stdout.write("\nInvalid token")
+                        exit(1)
+
+                self.token = self.args[index]
+                self.skip_next_arg = True
```

Comment:
What I'm realizing now is that the `--dir` arg uses a `=` and here we aren't using the same syntax

Reply:
I kept your same syntax I just moved the function into the class.